framework/src/engine/bft/method.ts

```diff
@@ -218,13 +222,21 @@ export class BFTMethod {
 			throw new Error('Invalid certificateThreshold input.');
 		}
 
-		sortValidatorsByBLSKey(validators);
+		const validatorsWithBFTWeight = validators
```

Comment:
probably we can add a comment on top of this to be careful in the future `Prepare a separate sorted list of validators for computing validatorsHash without modifying the existing validators array`